sdk/Lusid.Sdk/Utilities/ApiRetryHandler.cs

```diff
@@ -0,0 +1,68 @@
+
+namespace Lusid.Sdk.Utilities
+{
+    /// <summary>
+    /// Class used to define API error retry rules for all API calls.
+    /// </summary>
+    public static class ApiRetryHandler
+    {
+        /// <summary>
+        /// Number of max retry attempts
+        /// </summary>
+        public const int MaxRetryAttempts = 3;
+
+        /// <summary>
+        /// Get the internal exception condition on which to retry.
+        /// </summary>
+        /// <param name="restResponse">Response object that comes from the API Client</param>
+        /// <returns></returns>
+        private static bool GetInternalExceptionRetryCondition(IRestResponse restResponse)
+        {
+            return restResponse.ErrorException != null || restResponse.StatusCode == 0;
+        }
+
+        /// <summary>
+        /// Define Polly retry policy for synchronous API calls. Handles internal SDK exceptions only.
+        /// Use .Wrap() method to combine this policy with your other custom policies
+        /// </summary>
+        public static Policy<IRestResponse> GetInternalExceptionRetryPolicy() =>
+            Policy
+                .Handle<SystemException>()
+                .OrResult<IRestResponse>(GetInternalExceptionRetryCondition)
+                .Retry(
+                    MaxRetryAttempts,
+                    onRetry: (result, retryCount, context) =>
+                    {
+                        Console.WriteLine("An internal exception has occurred. Retrying. " +
+                                          $"Retry attempt: {retryCount}. Max retry attempts: {MaxRetryAttempts}");
+
+                        Console.WriteLine(
+                            "Temporarily logging the exception stack trace for finding the underlying root cause of this exception:\n" +
+                            $"{result.Result.ErrorException}\n");
+                    });
+
+
+        /// <summary>
+        /// Define Polly retry policy for asynchronous API calls. Handles internal SDK exceptions only.
+        /// Use .Wrap() method to combine this policy with your other custom policies
+        /// </summary>
+        public static AsyncPolicy<IRestResponse> GetInternalExceptionAsyncRetryPolicy() =>
+            Policy
+                .Handle<SystemException>()
+                .OrResult<IRestResponse>(GetInternalExceptionRetryCondition)
+                .RetryAsync(
+                    MaxRetryAttempts,
+                    onRetry: (result, retryCount, context) =>
+                    {
+                        Console.WriteLine("An internal exception has occurred. Retrying. " +
+                                          $"Retry attempt: {retryCount}. Max retry attempts: {MaxRetryAttempts}");
+
+                        Console.WriteLine(
+                            "Temporarily logging the exception stack trace for finding the underlying root cause of this exception:\n" +
+                            $"{result.Result.ErrorException}\n");
+                    });
```

Comment:
This looks like code duplication, I suggest delagating this and the one above in the Sync handler to a separate method.

Reply:
yeah that's fair